EIPS/eip-3403.md

```diff
@@ -59,7 +60,53 @@ TBD
 
 ## Test Cases
 
-TBD
+### Berlin costs
+
+With Berlin, and [EIP-2929](https://eips.ethereum.org/EIPS/eip-2929), the gas costs changes. Note, there is a difference between 'hot' and 'cold' slots.
+This table show the values assuming that all touched storage slots were already 'hot' (the difference being a one-time cost of 2000 gas).
+
+| Code | Used Gas | Refund | Original | 1st | 2nd | 3rd | Effective gas (after refund)
+| -- | -- | -- | -- | -- | -- | -- | -- | 
+| `0x60006000556000600055` | 212 | 0| 0 | 0 |  0 |  |  212 |
+| `0x60006000556001600055` | 20112 | 0| 0 | 0 |  1 |  |  20112 |
+| `0x60016000556000600055` | 20112 | 19900| 0 | 1 |  0 |  |  212 |
+| `0x60016000556002600055` | 20112 | 0| 0 | 1 |  2 |  |  20112 |
+| `0x60016000556001600055` | 20112 | 0| 0 | 1 |  1 |  |  20112 |
+| `0x60006000556000600055` | 3012 | 15000| 1 | 0 |  0 |  |  -11988 |
+| `0x60006000556001600055` | 3012 | 2800| 1 | 0 |  1 |  |  212 |
+| `0x60006000556002600055` | 3012 | 0| 1 | 0 |  2 |  |  3012 |
+| `0x60026000556000600055` | 3012 | 15000| 1 | 2 |  0 |  |  -11988 |
+| `0x60026000556003600055` | 3012 | 0| 1 | 2 |  3 |  |  3012 |
+| `0x60026000556001600055` | 3012 | 2800| 1 | 2 |  1 |  |  212 |
+| `0x60026000556002600055` | 3012 | 0| 1 | 2 |  2 |  |  3012 |
+| `0x60016000556000600055` | 3012 | 15000| 1 | 1 |  0 |  |  -11988 |
+| `0x60016000556002600055` | 3012 | 0| 1 | 1 |  2 |  |  3012 |
+| `0x60016000556001600055` | 212 | 0| 1 | 1 |  1 |  |  212 |
+| `0x600160005560006000556001600055` | 40118 | 19900| 0 | 1 |  0 |  1 |  20218 |
+| `0x600060005560016000556000600055` | 5918 | 17800| 1 | 0 |  1 |  0 |  -11882 |
+
+### With EIP-3403 partial refunds 
+
+If refunds were to be partially removed, as specified [here](https://github.com/ethereum/EIPs/pull/3403/), this would be the comparative table
```

Comment:
```suggestion
If refunds were to be partially removed, as specified here, this would be the comparative table
```
It is a bit odd for a page to link to itself like this, so I recommend removing the link.